### DISCUSSION TITLE
quickAddApi question: how to get the inputPrompt value in copy format?

### DISCUSSION THREAD

**Discussion by terrychenzw** [RyotaUshio/obsidian-pdf-plus#206] [Q&A]
I really love pdf++ plugin. But when i tried to use quickAddApi in link copy format, i met below problem.

https://github.com/RyotaUshio/obsidian-pdf-plus/assets/114856910/30566d05-d8c2-421f-b007-ddbb2111f4c3

As you saw in above vedio, the input value of the quickAdd inputPrompt can't be pasted. it returns "[object Promise]". 

So, I would like to ask some guidances/docs how to use quickAddApi in pdf++.

my link copy format as below
```js
{{linkWithDisplay}} {{quickAddApi.inputPrompt("输入一个字符")}}
```

**Comment by RyotaUshio** [comment]
Sorry for the late reply.
As stated in QuickAdd's docs (https://quickadd.obsidian.guide/docs/QuickAddAPI#inputpromptheader-string-placeholder-string-value-string-promisestring), `quickAddApi.inputPrompt` is asynchronous, i.e. it returns a `Promise` object.

> This function is asynchronous. You should await it.

However, PDF++ currently does not support `await` inside `{{...}}`, and probably will not in the future for some reasons.

The good news is that on my roadmap, I have this: https://github.com/RyotaUshio/obsidian-pdf-plus/issues/159

I will allow `async/await` in user scripts so please wait until the user script support is implemented.
